news/posts/DoodleBUGS-Introduction/index.qmd

```diff
@@ -0,0 +1,252 @@
+
+**Frontend architecture (Vue 3 + Cytoscape.js)**
+
+- Core graph state is managed in Vue; [Cytoscape.js](https://js.cytoscape.org/) handles layout, hit-testing, and interaction semantics (including compound nodes for plates) [@cytoscapejs].
+- Code generation lives in `DoodleBUGS/src/composables/useBugsCodeGenerator.ts` and maps `GraphNode`/`GraphEdge` to BUGS:
+  - Kahn topological sort for definition order
+  - Plate-aware recursion for `for (...) { ... }` blocks
+  - Parameter canonicalization (indices, numeric/expr passthrough)
+- Standalone Julia script generation uses `generateStandaloneScript()` in the same composable, mirroring backend execution.
+
+**Backend architecture (Julia)**
+
+- `run_model_handler()` in `DoodleBUGS/runtime/server.jl` materializes `model.bugs`, `payload.json`, and a transient `run_script.jl` that:
+  - Builds `NamedTuple`s from JSON or string-literal data/inits
+  - Compiles via `JuliaBUGS.@bugs`, wraps with `ADgradient(:ReverseDiff)` [@ReverseDiff]
+  - Samples with `AdvancedHMC.NUTS` through `AbstractMCMC` (Threads or Serial) [@AdvancedHMC; @AbstractMCMC; @HoffmanGelman2014]
+  - Emits summaries (`MCMCChains`, `DataFrames`) and quantiles to JSON
+    [@MCMCChains; @DataFrames]
+- Timeout: worker process is killed if exceeding `timeout_s`.
+- Cleanup: `safe_rmdir()` retries with GC to avoid EBUSY on Windows.
+
+## Why Vue (not React)?
+
+The proposal planned React; we chose Vue 3 after evaluating the graph layer and developer velocity for this app.
+
+- Tried Konva (canvas) for custom graph editing: powerful drawing primitives, but required bespoke graph semantics (hit testing, edge routing, compound nodes) that [Cytoscape.js](https://js.cytoscape.org/) provides out of the box.
+- Tried D3 force/layouts: flexible, but compound nodes (plates), nesting, and drag constraints became a significant amount of custom code to maintain.
+- [Cytoscape.js](https://js.cytoscape.org/) offered:
+  - Native graph model with compound nodes (great for plates)
+  - Integrated layouts (WebCola, KLay) and rich interaction APIs [@webcola; @elk]
+  - Mature ecosystem and performance characteristics for medium-sized graphs
+- [Vue 3](https://vuejs.org/) (vs React) for this project:
+  - Composition API made integrating an imperative graph library (Cytoscape) straightforward via composables and lifecycle hooks
+  - SFC ergonomics and Pinia stores enabled quick iteration with strong TypeScript support
+  - Template reactivity + refs reduced reconciliation overhead when bridging to Cytoscape’s imperative API
+  - Minimal glue code for state management (Pinia) vs setting up reducers/selectors; enabled rapid iteration
+  - Vite + Vue tooling yielded fast HMR for UI-heavy iterations
+- Design inspirations: draw.io for interaction affordances; Stan Playground for model/run UX [@drawio; @stan-playground].
+
+## Comparison to Legacy DoodleBUGS
```

Comment:
It's useful to have a few screenshots to compare the legacy DoodleBUGS with the new one.